### PR TITLE
Unit tests for networking

### PR DESCRIPTION
**Pull request by greatman**
Doing unit tests for networking. Should prevent stupid bugs between encode/decode.

### CODE REVIEW

**Comment by turt2live** [comment]
@greatman Looks like this fails to compile. This PR is also still marked as a WIP. If this is ready for full review, please remove the WIP tag from the title. Otherwise, please indicate work still needs to be completed (and what that is).

The PR should be able to be compiled for further review. Thanks!

**Comment by dequis** [comment]
More checkstyle failures lol

**Comment by greatman** [comment]
@turt2live Totally not ready :P

**Comment by greatman** [comment]
Due to a derp in my rebasing it will need my PR #142 since I got those packets in.

**Comment by Wild1145** [comment]
There is some formatting that I think needs correcting, the bit im most unsure about is src/main/java/net/glowstone/net/codec/play/player/CombatEventCodec.java line 21, it might be fine, but if someone who knows a fair bit about the way the project style works could just double check, it looks like it only needs a break put in.

Im also not sure on the way that src/test/java/net/glowstone/net/BaseProtocolTest.java line 57 has been formatted, it might be fine, but its just something maybe someone with more experience in this code style wouldnt mind just taking a quick look at, might just need a break added in. 

There is also whitespace added in src/main/java/net/glowstone/net/message/KickMessage.java line 15 that doesnt need to be there. 

Looks to be a extra blank line @ src/main/java/net/glowstone/net/message/play/player/InteractEntityMessage.java line 79

 src/test/java/net/glowstone/net/LoginProtocolTest.java line 14 is just a extra blank line that can be removed. 

src/test/java/net/glowstone/net/PlayProtocolTest.java line 75 can be removed also, it is another blank line that shouldn't be there

I hope this is still easy to find what Im talking about and is okay to be posting,... I know that commenting on each line will just start to mass spam you, so with any luck this is just as informal :)

**Comment by dequis** [comment]
Reviewing stuff _other_ than whitespace would be useful, you know. At this point, if checkstyle doesn't complain about it, then it's okay.

**Comment by Wild1145** [comment]
Fair enough. Just posting what I can be somewhat sure about.

Also, the Travis CI cant build this PR...

**Comment by greatman** [comment]
The tests arent ready, hence travis failing ;)

**Comment by greatman** [comment]
Now ready!

File: src/main/java/net/glowstone/entity/meta/MetadataIndex.java [hunk 1]
```diff
@@ -114,6 +114,18 @@ private MetadataIndex(int index, MetadataType type, Class<? extends Entity> appl
         this.appliesTo = appliesTo;
     }
 
+    //TODO Ugly hack so unit tests can read metadata
```
Comment (dequis) [inline src/main/java/net/glowstone/entity/meta/MetadataIndex.java:117]: Nothing says "ready" like a TODO in the first line of the diff

Comment (greatman) [inline src/main/java/net/glowstone/entity/meta/MetadataIndex.java:117]: Well it's more of a "If something is better, please implement it" and I made it TODO so it shows in the IDE.

**Comment by SpaceManiac** [comment]
Merged in 433d7bf3d94af5b4ee85261b272c22d482b73462. Thanks!

**Pull request closed without merging**